Home Work #2

DeeSaiZzZ opened the pull request:
1. Implements all functional.
2. Added properties file.
3. Implement BeanFactoryPostProcessor and BeanPostProcessor.

Nazarii-Martynyshyn-EPAM committed:
1. Implements all functional.
2. Added properties file.
3. Implement BeanFactoryPostProcessor and BeanPostProcessor.

DeeSaiZzZ committed:
Update application.properties

DeeSaiZzZ committed:
Update pom.xml

Nazarii-Martynyshyn-EPAM committed:
Refactoring

KostyantynPanchenko commented on src/main/java/com/epam/spring/homework2/Main.java:
```diff
@@ -0,0 +1,36 @@
+
+        System.out.println(">>> All beans: ");
+        Arrays.stream(appContext.getBeanDefinitionNames())
+                .forEach(System.out::println);
```
Також потрібно було зробити вивести `appContext.getBeanDefinition(beanDefinitionName)`

KostyantynPanchenko commented on src/main/java/com/epam/spring/homework2/beans/BeanEntity.java:
```diff
@@ -0,0 +1,33 @@
+    protected int value;
+
+    public BeanEntity() {
+
```
Який це convention? Так не роблять - порожній рядок в тілі конструктора не залишають

KostyantynPanchenko commented on src/main/java/com/epam/spring/homework2/beans/BeanEntity.java:
```diff
@@ -0,0 +1,33 @@
+
+public abstract class BeanEntity {
+
+    protected String name;
```
`private` + getter (за потреби) а не `protected`

KostyantynPanchenko commented on src/main/java/com/epam/spring/homework2/configs/FirstConfig.java:
```diff
@@ -0,0 +1,39 @@
+public class FirstConfig {
+
+    @Autowired
+    public Environment env;
```
Спосіб робочий, разомз тим, я б очікував більш елегантний і спеціалізований спосіб вставити значення із application.properties файлу, зокрема:

```java
@Bean(initMethod = "initMethod", destroyMethod = "destroyMethod")
@DependsOn("beanD")
public BeanB beanB(@Value("${beanB.name}") final String name, @Value("${beanB.value}") final int value) {
    return new BeanB(name, value);
 }
```

Nazarii-Martynyshyn-EPAM committed:
1. Add getBeanDefinition() by name
2. Fill empty constructor in class BeanEntity
3. Make fields (name, value) private, added getters and setters
4. BeanB, BeanC, BeanD initialize by annotation Value
